core/src/main/java/com/sequenceiq/cloudbreak/controller/validation/environment/EnvironmentCreationValidator.java

```diff
@@ -87,4 +96,14 @@ private void validateKubernetesConfigs(Environment subject, EnvironmentV4Request
                     String.join(", ", requestedKubernetesConfigs)));
         }
     }
+
+    private void validateNetwork(EnvironmentV4Request request, String cloudPlatform, ValidationResultBuilder resultBuilder) {
+        EnvironmentNetworkV4Request networkV4Request = request.getNetwork();
+        if (networkV4Request != null) {
+            EnvironmentNetworkValidator environmentNetworkValidator = environmentNetworkValidatorsByCloudPlatform.get(CloudPlatform.valueOf(cloudPlatform));
```

Comment:
this can be shortened with `computeIfAbsent` like this: `environmentNetworkValidatorsByCloudPlatform.computeIfAbsent(CloudPlatform.valueOf(cloudPlatform), key -> { throw new BadRequestException(key + " platform is not supported"); });`

Reply:
I think the if statement is easier to read/understand in this case.

Reply:
Alright

Reply:
But shouldn't the validator throw an exception if an unsupported cloud provider comes in the request?

Reply:
Yes, it make sense to leave an error through ValidationResultBuilder, the modification on the go!